pkg/ddc/alluxio/transform.go

```diff
@@ -403,82 +410,63 @@ func (e *AlluxioEngine) allocatePorts(value *Alluxio) error {
-	//
-	//// if len(allocatedPorts) == PORT_NUM {
-	//value.Master.Ports.Rpc = allocatedPorts[0]
-	//value.Master.Ports.Web = allocatedPorts[1]
-	//value.Worker.Ports.Rpc = allocatedPorts[2]
-	//value.Worker.Ports.Web = allocatedPorts[3]
-	//value.JobMaster.Ports.Rpc = allocatedPorts[4]
-	//value.JobMaster.Ports.Web = allocatedPorts[5]
-	//value.JobWorker.Ports.Rpc = allocatedPorts[6]
-	//value.JobWorker.Ports.Web = allocatedPorts[7]
-	//value.JobWorker.Ports.Data = allocatedPorts[8]
-	//
-	//index := 9
-	//if e.runtime.Spec.APIGateway.Enabled {
-	//	if len(allocatedPorts) <= index {
-	//		e.Log.Info("the number of port check failed, when api gateway is enabled",
-	//			"PORT_NUM", index+1,
-	//			"len(allocatedPorts)", len(allocatedPorts),
-	//			"allocatedPorts", allocatedPorts)
-	//		return fmt.Errorf("The lengh of port list is %v less than expected %v", len(allocatedPorts), PORT_NUM)
-	//	}
-	//	value.APIGateway.Ports.Rest = allocatedPorts[index]
-	//}
-	//
-	//if e.runtime.Spec.Master.Replicas > 1 {
-	//	if len(allocatedPorts) <= index+2 {
-	//		e.Log.Info("the number of port check failed, when Embedded is enabled",
-	//			"index", index+2,
-	//			"len(allocatedPorts)", len(allocatedPorts),
-	//			"allocatedPorts", allocatedPorts)
-	//		return fmt.Errorf("The lengh of port list is %v less than expected %v", len(allocatedPorts), PORT_NUM)
-	//	}
-	//	value.Master.Ports.Embedded = allocatedPorts[index+1]
-	//	value.Master.Ports.Embedded = allocatedPorts[index+2]
-	//}
-	//
-	//// } else {
-	//// 	value.Master.Ports.Embedded = allocatedPorts[9]
-	//// 	value.JobMaster.Ports.Embedded = allocatedPorts[10]
-	//// }
-
 	return nil
 }
 
-// // 8.set default port for fluid engine
-// func (e *AlluxioEngine) setDefaultPorts(value *Alluxio) {
-// 	if e.runtime.Spec.Master.Replicas > 1 {
-// 		value.Master.Ports.Rpc = 19998
-// 		value.Master.Ports.Web = 19999
-// 		value.Worker.Ports.Rpc = 29999
-// 		value.Worker.Ports.Web = 30000
-// 		value.JobMaster.Ports.Rpc = 20001
-// 		value.JobMaster.Ports.Web = 20002
-// 		value.JobWorker.Ports.Rpc = 30001
-// 		value.JobWorker.Ports.Web = 30003
-// 		value.JobWorker.Ports.Data = 30002
-// 	} else {
-// 		value.Master.Ports.Embedded = 19200
-// 		value.JobMaster.Ports.Embedded = 20003
-// 	}
-// }
-
 func (e *AlluxioEngine) transformMasterSelector(runtime *datav1alpha1.AlluxioRuntime) map[string]string {
 	properties := map[string]string{}
 	if runtime.Spec.Master.NodeSelector != nil {
 		properties = runtime.Spec.Master.NodeSelector
 	}
 	return properties
 }
+
+func (e *AlluxioEngine) transformJournal(runtime *datav1alpha1.AlluxioRuntime, value *Alluxio) {
+	journalType, found := runtime.Spec.Properties["alluxio.master.journal.type"]
+
+	if !found {
+		journalType = JOURNAL_TYPE_UFS
+	}
+	volumeType := runtime.Spec.Master.Journal.VolumeType
+	if volumeType == "" {
+		volumeType = "memory"
+	}
+
+	storageClass := runtime.Spec.Master.Journal.StorageClass
+	if storageClass == "" {
+		storageClass = "standard"
```

Comment:
It is best to define `memory ` and `standard ` in the same place where including all the default values

The advantage is that when we upgrade later, we know how much of the default action was done, and what was the default action